shared/src/main/scala/mlscript/codegen/typescript/TsTypegen.scala

```diff
@@ -164,13 +429,21 @@ final class TsTypegenCodeBuilder {
       // these types may mutate typegen context by argCounter, or
       // by creating new type aliases
       case f@Function(lhs, rhs) =>
-        val arg = typegenCtx.termScope.declareRuntimeSymbol("arg");
-
         // flip polarity for input type of function
         // lhs translates to the complete argument list
-        val lhsTypeSource = toTsType(lhs, true)(typegenCtx, pol.map(!_));
-        val rhsTypeSource = toTsType(rhs);
+        // method definition only affects source code representation of top level function
+        val lhsTypeSource = lhs match {
+          // tuple are handled specially because
+          // they create their own arg names
+          case Tuple(fields) => toTsType(lhs, true)(typegenCtx, pol.map(!_))
+          // there are other types which might hide tuple arguments
+          // like recursive types, AppliedTypes or bounded types
+          // their tuple types can be uniformly converted
+          // using a spread operator
+          case _ => (SourceCode("...arg: ") ++ toTsType(lhs, true)(typegenCtx, pol.map(!_))).parenthesized
```

Comment:
It seems this section of code is an explicit duplicate of the one above. Wouldn't it be better to factor them out into a helper function? Otherwise they may diverge later although they're meant to do the same thing.

Reply:
Yes fixed it.